lib/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.22...3.25)
+
+legacy_check()
+
+add_library(obs-websocket-api INTERFACE)
+add_library(OBS::websocket-api ALIAS obs-websocket-api)
+
+target_sources(obs-websocket-api INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket-api.h>
+                                           $<INSTALL_INTERFACE:${OBS_INCLUDE_DESTINATION}/obs-websocker-api.h>)
```

Comment:
This doesn't seem correct to me - the source list should just be `obs-websocket-api.h` in all situations - the path is resolved by `target_include_directories` below, where a differentiation between build interface and install interface is needed.

Reply:
I just followed those explanations:

https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/

> The paths used for the project’s own build will be specific to the machine on which that build is performed and to the directory in which it is built. Once installed, those directories will likely not be accessible, so the paths would need to be replaced with something else that makes sense for the installed set of files. This is made possible using the BUILD_INTERFACE and INSTALL_INTERFACE generator expressions. The following example shows how to provide different paths for the same file in the two different contexts:
>
>```cmake
>target_sources(myHeaderOnly
>    PUBLIC
>        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/algo.h>
>        $<INSTALL_INTERFACE:include/algo.h>
>)
>install(FILES algo.h DESTINATION include)
>```

Reply:
Yeah that advice is for old CMake versions and isn't necessary anymore ever since CMake gained the ability to automatically treat sources added to a target as relative to the source directory (changed in CMake 3.13, released in 2018).

The only place where that differentiation is needed is when the sources added use an absolute path.

Otherwise just specifying a different include directory based on build or install interface is sufficient (see also how we handle public headers in `libobs`).